### PR TITLE
Added Code (1)

### PR DESCRIPTION
**Pull request by mmarchetti2**
I'm pretty sure I added the correct code, as it seemed to work properly.

### CODE REVIEW

**Comment by mmarchetti2** [comment]
Um, okay, I guess it's wrong? Can you tell what the problem is?

File: randomQuizGenerator.py [hunk 1]
```diff
@@ -23,11 +23,36 @@
...
-            
-            
\ No newline at end of file
+#TODO: follow the 'generating random quiz files' project in the textbook to fill in this file.
+#TODO: however, make the following modificatiosn to the instructions on the textbook:
+#       1. instead of making 35 quiz versions, you'll only make 5 quiz versions
+#       2. instead of creating quiz and answer files in the current working directory, create a folder titled 'quizzes' and another folder titled 'answers'.
+#       3. place the randomly-generated quizzes in the 'quizzes' directory.
+#       4. plaec the corresponding answers in the 'answers' directory.
+os.makedirs('.\\quizzes')
```
Comment (ngocnguyen83) [inline randomQuizGenerator.py]: Hi, assume that all other codes are correct, try to change " .\\ " to "./" for these two lines.

File: randomQuizGenerator.py [hunk 2]
```diff
@@ -23,11 +23,36 @@
...
+os.makedirs('.\\answers')
+
+for quizNum in range(5):
+    quizFile = open('.\\quizzes\\capitalsquiz%s.txt' % (quizNum + 1), 'w')
```
Comment (ngocnguyen83) [inline randomQuizGenerator.py]: for line 36 and 37, also try to change " .\\ " to " ./ " AND " \\ " to " / " . I hope it works.

**Comment by mmarchetti2** [comment]
I think it should work, but apparently there's still something wrong with the code... Thanks for the suggestion, by the way. :D

**Comment by halfaro1** [comment]
Great work collaborating, @mmarchetti2 and @ngocnguyen83 
I think you're on the right track, but the issue isn't double slashes, it's that the code is written for Windows and the automated checking is happening on a Linux machine. To help with this, use the `os.join` function described in the textbook.

**Comment by mmarchetti2** [comment]
Um, okay, apparently that wasn't the correct usage, I'll try it again.

**Comment by mmarchetti2** [comment]
Okay, this is a bit of a test, so I'll see if it works.

**Comment by mmarchetti2** [comment]
Hm... well I guess I'm clueless as to what to do. I've tried several different uses, but none seem to pass the checks. How would I use os.path.join to change the .\\ commands? I think that's the main problem, and I can't really find the solution.

**Comment by mmarchetti2** [comment]
Okay, so far all of these tests work on my computer and it's even doing the .\\ correctly. What else do I have to do? Is there something wrong that doesn't have to do with the file directories? Any help would be awesome. :)

File: randomQuizGenerator.py [hunk 3]
```diff
@@ -29,12 +29,12 @@
...
 #       4. plaec the corresponding answers in the 'answers' directory.
-os.makedirs(os.path.join('quizzes'))
-os.makedirs(os.path.join('answers'))
+os.makedirs(os.path.join('.','quizzes'))
+os.makedirs(os.path.join('.','answers'))
 
 for quizNum in range(5):
-    quizFile = open(os.path.join('quizzes','capitalsquiz%s.txt' % (quizNum + 1)), 'w')
-    answerKeyFile = open(os.path.join('answers','capitalsquiz_answers%s.txt' % (quizNum + 1)), 'w')
+    quizFile = open(os.path.join('.','quizzes','capitalsquiz%s.txt' % (quizNum + 1)), 'w')
```
Comment (halfaro1) [inline randomQuizGenerator.py]: @mmarchetti2 you don't need the period, but you have good usage of the join function here

**Comment by mmarchetti2** [comment]
Hm... I hate that I can't seem to get this right. Whatever the case, do I just need the periods for the makedirs commands? Or am I not supposed to use the period at all? At any rate, I don't remember them giving this type of scenario in the book. Thank you, again, for your patience and help. Oh, and Happy Easter!

**Comment by halfaro1** [comment]
@mmarchetti2 you don't need the periods if the directories live in the current working directory

**Comment by mmarchetti2** [comment]
Hm... TestChange(3) had no periods and it still didn't pass. Well, I'll try it again just in case. I can't seem to find anything else wrong with the code. It's working perfectly fine on my computer.

File: randomQuizGenerator.py [hunk 4]
```diff
@@ -23,11 +23,36 @@
...
-            
-            
\ No newline at end of file
+#TODO: follow the 'generating random quiz files' project in the textbook to fill in this file.
+#TODO: however, make the following modificatiosn to the instructions on the textbook:
+#       1. instead of making 35 quiz versions, you'll only make 5 quiz versions
+#       2. instead of creating quiz and answer files in the current working directory, create a folder titled 'quizzes' and another folder titled 'answers'.
+#       3. place the randomly-generated quizzes in the 'quizzes' directory.
+#       4. plaec the corresponding answers in the 'answers' directory.
+os.makedirs(os.path.join('quizzes'))
```
Comment (halfaro1) [inline randomQuizGenerator.py]: @mmarchetti2 The `os.path.join` function is only necessary if you are joining directories, or files within directories. In this case, the `quizzes` directory is standalone so you don't need it.

File: randomQuizGenerator.py [hunk 5]
```diff
@@ -23,11 +23,36 @@
...
+os.makedirs(os.path.join('answers'))
+
+for quizNum in range(5):
+    quizFile = open(os.path.join('quizzes','capitalsquiz%s.txt' % (quizNum + 1)), 'w')
```
Comment (halfaro1) [inline randomQuizGenerator.py]: @mmarchetti2 the closing parentheses for your `os.path.join` function should follow the end of the file name, in this case, the closing parentheses should be after `, 'capitals quiz%s.txt'`

**Comment by mmarchetti2** [comment]
Oh, that makes sense. It stinks that I didn't find that out myself. Now that I look at it, it really makes sense.

**Comment by mmarchetti2** [comment]
...Wow. And yet, something is still wrong. This is just embarrassing.

**Comment by halfaro1** [comment]
@mmarchetti2 does this run on your computer? The automated testing is supposed to remove the directories `answers` and `quizzes` if they already exist. The reasoning is that if you are turning in a project with these directories already created, the test script will remove them so that it can run your script from scratch to test them again. It looks like it's tripping up when it's trying to remove those folders since you're turning in files that don't already have those folders.

**Comment by mmarchetti2** [comment]
Oh, that would explain it. First off, yes, it runs perfectly fine on my computer. And secondly, I always deleted the 'answer' and 'quizzes' folders before I submitted it. I'll try it again.

**Comment by mmarchetti2** [comment]
Wow, that was it? I guess that's a bit anti-climactic. Oh well, thanks for the help!

**Comment by halfaro1** [comment]
@mmarchetti2 Wohoo! Glad it worked and I'm sorry -- that's my fault for a badly written test script, not yours. I'll have to refine my script for future students. If you want a little look under the hood, take a look at the first few lines of the test file. To correct this, I'll wrap the calls to the shutil package in a try/except block. That will catch any errors generated if a student doesn't already have the folders already created. 

I'll grade this soon and get your grade on blackboard by Friday.

**Comment by mmarchetti2** [comment]
Okay, I'm glad something good came of all this :D
Whatever the case, thanks again!

**Comment by halfaro1** [comment]
@mmarchetti2 
Fantastic work :+1:
I'm marking this pull request as closed, and labeling it as `graded` while I finish grading these projects. When I'm done grading all the projects, and your grade should be on Blackboard, I'll change the tag to `grade updated on blackboard`

**Pull request closed without merging**